src/bin/inx-chronicle/cli.rs

```diff
@@ -33,4 +41,98 @@ pub struct ClArgs {
+            .map(ChronicleConfig::from_file)
+            .transpose()?
+            .unwrap_or_default();
+
+        if let Some(conn_str) = &self.mongodb_conn_str {
+            config.mongodb.conn_str = conn_str.clone();
+        }
+
+        #[cfg(all(feature = "stardust", feature = "inx"))]
+        {
+            if let Some(connect_url) = &self.inx_addr {
+                config.inx.connect_url = connect_url.clone();
+            }
+            if let Some(enabled) = self.enable_inx {
+                config.inx.enabled = enabled;
+            }
+        }
+
+        #[cfg(feature = "api")]
+        {
+            if let Some(password) = &self.password {
+                config.api.password_hash = hex::encode(
+                    argon2::hash_raw(
+                        password.as_bytes(),
+                        config.api.password_salt.as_bytes(),
+                        &Into::into(&config.api.argon_config),
+                    )
+                    // TODO: Replace this once we switch to a better error lib
+                    .expect("invalid JWT config"),
+                );
+            }
+            if let Some(path) = &self.identity_path {
+                config.api.identity_path.replace(path.clone());
+            }
+            if let Some(enabled) = self.enable_api {
+                config.api.enabled = enabled;
+            }
+        }
+
+        if let Some(enabled) = self.enable_metrics {
+            config.metrics.enabled = enabled;
+        }
+
+        Ok(config)
+    }
+
+    /// Process subcommands and return whether the app should early exit.
+    #[allow(unused)]
+    #[allow(clippy::collapsible_match)]
+    pub fn process_subcommands(&self, config: &ChronicleConfig) -> Result<bool, Error> {
+        if let Some(subcommand) = &self.subcommand {
+            match subcommand {
+                #[cfg(feature = "api")]
+                Subcommands::GenerateJWT => {
+                    use crate::api::ApiData;
+                    let api_data = ApiData::try_from(config.api.clone()).expect("invalid API config");
+                    let claims = auth_helper::jwt::Claims::new(
+                        ApiData::ISSUER,
+                        uuid::Uuid::new_v4().to_string(),
+                        ApiData::AUDIENCE,
+                    )
+                    .unwrap() // Panic: Cannot fail.
+                    .expires_after_duration(api_data.jwt_expiration)
+                    .map_err(crate::api::ApiError::InvalidJwt)?;
+                    let exp_ts = time::OffsetDateTime::from_unix_timestamp(claims.exp.unwrap() as _).unwrap();
+                    let jwt = auth_helper::jwt::JsonWebToken::new(claims, api_data.secret_key.as_ref())
+                        .map_err(crate::api::ApiError::InvalidJwt)?;
+                    println!("Bearer {}", jwt);
+                    println!("Expires: {}", exp_ts);
+                    return Ok(true);
+                }
+                _ => (),
```

Comment:
Now that we have error types, can't we return something like `unknown subcommand` here?